ui/e2e/instances/details/index.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { createNamespace, deleteNamespace } from "../../utils/namespace";
+import { expect, test } from "@playwright/test";
+
+import { createFile } from "e2e/utils/files";
+import { createInstance } from "../utils/index";
+import { faker } from "@faker-js/faker";
+import { simpleWorkflow as simpleWorkflowContent } from "../utils/workflows";
+
+let namespace = "";
+const simpleWorkflowName = faker.system.commonFileName("yaml");
+
+test.beforeEach(async () => {
+  namespace = await createNamespace();
+  /* create workflows we can use to create instances later */
+  await createFile({
+    name: simpleWorkflowName,
+    namespace,
+    type: "workflow",
+    yaml: simpleWorkflowContent,
+  });
+});
+
+test.afterEach(async () => {
+  await deleteNamespace(namespace);
+  namespace = "";
+});
+
+test("the header of the instance page shows the relevant data for the workflow", async ({
+  page,
+}) => {
+  const newInstance = createInstance({ namespace, path: simpleWorkflowName });
+
+  await expect(newInstance, "wait until process was completed").toBeDefined();
+
+  const instanceId = (await newInstance).instance;
+
+  await page.goto(`/${namespace}/instances/${instanceId}`);
+
+  const header = page.getByTestId("instance-header-container");
+  await expect(header, "It renders the header").toBeVisible();
+
+  const instanceIdHeader = header.locator("h3");
+  await expect(
+    instanceIdHeader,
+    "It renders the instance ID in the header"
+  ).toBeVisible();
+
+  const instanceIdNavLink = page.locator("ul").locator("a").nth(2);
+  await expect(
+    instanceIdNavLink,
+    "It renders the instance ID in the breadcrumb navigation"
+  ).toBeVisible();
+
+  expect(instanceIdHeader.innerText, "the instance IDs are the same").toEqual(
+    instanceIdNavLink.innerText
+  );
```

Comment:
I still think we need to explicitly check the IDs here. At this point, we know, that the text of the last breadcrumb segment and the headline are the same. 

My suggested solution

```typescript
  const instanceIdShort = instanceId.slice(0, 8);
  await expect(
    header.locator("h3"),
    "It renders the instance ID in the header"
  ).toHaveText(instanceIdShort);

  await expect(
    page.locator("ul").locator("a").nth(2),
    "It renders the instance ID in the breadcrumb navigation"
  ).toHaveText(instanceIdShort);
```

Reply:
Ah I didn't get that. Yep, that's better.